CONTRIBUTING.md

```diff
@@ -25,3 +25,55 @@ All contributors to this project must have a signed Contributor License Agreemen
 We generally only need you (or your employer) to sign our CLA once and once signed, you should be able to submit contributions to any VMware project.
 
 Note: if you would like to submit an "_obvious fix_" for something like a typo, formatting issue or spelling mistake, you may not need to sign the CLA. Please see our information on [obvious fixes](https://cla.pivotal.io/about#obvious-fix) for more details.
+
+# Development
+
+## Prerequisites
+
+- [minikube](https://minikube.sigs.k8s.io/docs/)
+- [ytt](https://github.com/k14s/ytt)
+- [pack 0.8.1](https://github.com/buildpacks/pack)
+
+## Run Unit tests
+```bash
+# Run all tests
+./hack/test.sh
+# or run single test
+./hack/test.sh -run TestLogger
+```
+
+## Run E2E tests against minikube registry
+```bash
+# Bootstrap k8s cluster and enable docker registry
+# X.X.X.X must be replaced with your subnetmask of "minikube ip"
+minikube start --driver=docker --insecure-registry=X.X.X.X/16
+# Build kbld binary for testing
+./hack/build.sh
+# Make your env aware of the docker registry
+eval $(minikube docker-env)
+# Run all tests
+./hack/test-all-minikube-local-registry.sh
+# or run single test
+./hack/test-all-minikube-local-registry.sh -run TestDockerBuildSuccessful
+```
+
+## Run E2E tests against private docker registry
+```bash
+# Bootstrap k8s cluster
+minikube start
+# Build kbld binary for testing
+./hack/build.sh
+# Make your env aware of the docker registry
+eval $(minikube docker-env)
```

Comment:
I'm not sure but this should not be required when using the docker registry?

Reply:
i typically use minikube _because_ it comes with docker daemon (since i dont install docker daemon directly on my machine). if you just have docker directly on your machine then none of the minikube stuff is needed since kbld does not know anything about k8s.

Reply:
Good to know. Using docker hub is no good default because they have a fixed limit of 3 repositories for not paying customers (didn't work on my machine)